README.md

```diff
@@ -5,7 +5,65 @@ This repository contains the code to generate various versions of the Datadog Ag
 * https://s3.amazonaws.com/dd-agent/scripts/install_script_agent6.sh to install Agent 6
 * https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh to install Agent 7
 
-Usage instructions for the install script are in the [Datadog App](https://app.datadoghq.com/account/settings#agent/overview).
+
+## Agent install script usage
+Basic usage instructions for the agent install script are in the [Datadog App](https://app.datadoghq.com/account/settings#agent/overview).
+
+### Agent configuration options
+Install script allows installation of different flavours of the Agent binaries and allows automatic setting of some configuration options. We list them below with their diffent possible values. 
+
+> [!IMPORTANT]
+> All variables are optional except `DD_API_KEY`, which is required unless `DD_UPGRADE` (upgrade from datadog-agent 5) is set.
+
+> [!WARNING]
+> Install script input options are only considered at initial installation, and they don't overwrite pre-existing configuration files
+
+| Variable | Description|
+|-|-|
+|`DD_AGENT_FLAVOR`|The agent binary to install. Possible values are `datadog-agent`(default), `datadog-iot-agent`, `datadog-dogstatsd`, `datadog-fips-proxy`, `datadog-heroku-agent`.|
+|`DD_API_KEY`|The application key to access Datadog's programatic API.|
+|`DD_SITE`|The site of the Datadog intake to send Agent data to (e.g. `datadoghq.com`)|
+|`DD_URL`|The host of the Datadog intake server to send metrics to (e.g. `https://app.datadoghq.com`). Only set this option if you need the Agent to send metrics to a custom URL: it overrides the site setting defined in "site". It does not affect APM, Logs or Live Process intake which have their own "*_dd_url" settings.|
+|`DD_HOSTNAME`|Force the hostname name.|
+|`DD_HOST_TAGS`|List of host tags, defined as a comma-separated list of `key:value` strings (e.g. `team:infra,env:prod`) Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.|
+|`DD_UPGRADE`|Set to any value to trigger a version upgrade from datadog-agent 5. It will import configuration files from `/etc/dd-agent/datadog.conf`. Not compatible with `DD_AGENT_FLAVOR=datadog-dogstatsd`.|
+|`DD_FIPS_MODE`|Set to any value to enable the use of the FIPS proxy to send data to the DataDog backend. Enabling this will force all outgoing traffic from the Agent to the local proxy. It's important to note that enabling this will not make the Datadog Agent FIPS compliant, but will force all outgoing traffic to a local FIPS compliant proxy. By-pass `DD_SITE` and `DD_URL` when enabled|
```

Comment:
```suggestion
|`DD_FIPS_MODE`|Set to any value to enable the use of the FIPS proxy to send data to the DataDog backend. Enabling this option forces all outgoing traffic from the Agent to the local proxy. It's important to note that enabling this will not make the Datadog Agent FIPS compliant, but will force all outgoing traffic to a local FIPS compliant proxy. By-pass `DD_SITE` and `DD_URL` when enabled. For more information on FIPS compliance, see [FIPS Compliance](https://docs.datadoghq.com/agent/configuration/agent-fips-proxy/).|
```